### PR TITLE
Add likely_subtags: data generation, node executor

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

**Comment by sven-oly** [comment]
This uses the latest CLDR data in https://github.com/unicode-org/cldr/blob/main/common/testData/localeIdentifiers/likelySubtags.txt for checking Node's minimize and maximize locale tags.

File: testgen/testdata_gen.py [hunk 1]
```diff
@@ -172,6 +172,83 @@ def generateLanguageNameTestDataObjects(self, rawtestdata, json_tests, json_veri
...
+            # split at ";" and ignore whitespace
+            tags = list(map(str.strip, line.split(';')))
+            # This should give 3 items
+            if len(tags) < 3:
+                logging.warning('Likely subtags: count of tages != 3: %s',
+                                line)
+                continue
+
+            # Create 3 minimize tests
+            expected = tags[2]
```
Comment (sffc) [inline testgen/testdata_gen.py:211]: Suggestion: you should read `tags[3]` and put it in an option called `minimizeFavorScript`

File: testgen/testdata_gen.py [hunk 2]
```diff
@@ -172,6 +172,83 @@ def generateLanguageNameTestDataObjects(self, rawtestdata, json_tests, json_veri
...
+            for index in range(0, 3):
+                label = str(count).rjust(max_digits, '0')
+                locale = tags[index]
+                test_min = {'label': label,
+                         'locale': locale,
+                         'option': 'minimize'
+                         }
+                verify = {'label': label,
+                         'verify': expected
+                          }
```
Comment (sffc) [inline testgen/testdata_gen.py:221]: Please note the highlighted portions of the notice at the top of the test data file:

> Source ;	AddLikely ;	RemoveFavorScript ;	RemoveFavorRegion
>   Source: a locale to which the following operations are applied.
>   AddLikely: the result of the Add Likely Subtags.
>                      If Add Likely Subtags fails, then “FAIL”.
>   RemoveFavorScript: Remove Likely Subtags, when the script is favored.
>                      ***Only included when different than AddLikely.***
>   RemoveFavorRegion: Remove Likely Subtags, when the region is favored.
>                      ***Only included when different than RemoveFavorScript.***

I don't immediately see where you are doing the "only included when..." portion of the instructions. Most but not all cases have data populated for RemoveFavorScript, and only a few have data populated for RemoveFavorRegion

File: testgen/testdata_gen.py (hunk 1 above)
Comment (sven-oly) [reply testgen/testdata_gen.py:211]: Thanks. Will do!

File: testgen/testdata_gen.py (hunk 2 above)
Comment (sven-oly) [reply testgen/testdata_gen.py]: Updated tests and code in latest commits.

File: executors/node/likely_subtags.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    try {
+      let result_locale;
+      if (test_option === 'maximize') {
+        result_locale = intl_locale.maximize().baseName;
+      } else if (test_option === 'minimizeFavorScript' ||
+                 test_option === 'minimize') {
+        result_locale = intl_locale.minimize().baseName;
+      } else if (test_option === 'minimizeFavorRegion') {
+        result_locale = intl_locale.minimizeFavorRegion().baseName;
```
Comment (sffc) [inline executors/node/likely_subtags.js:33]: Note: there is no `Intl.Locale.prototype.minizeFavorRegion`; this should be unsupported

File: testgen/testdata_gen.py [hunk 4]
```diff
@@ -172,6 +172,103 @@ def generateLanguageNameTestDataObjects(self, rawtestdata, json_tests, json_veri
...
+            tags = list(map(str.strip, line.split(';')))
+            # This should give 3 items
+            if line.find('FAIL') >= 0:
+                # What do do with this?
+                pass
+
+            # Create minimize tests - default is RemoveFavorScript
+            if tags[2]:
+                remove_favor_script = tags[2]
+                for tag in tags:
```
Comment (sffc) [inline testgen/testdata_gen.py:212]: This generates extra cases for each line. Probably not needed.

File: testgen/testdata_gen.py (hunk 2 above)
Comment (sven-oly) [reply testgen/testdata_gen.py]: Updated to handle non-included values, falling back to previous.

File: testgen/testdata_gen.py (hunk 1 above)
Comment (sven-oly) [reply testgen/testdata_gen.py]: Done!

File: testgen/testdata_gen.py [hunk 5]
```diff
@@ -172,6 +172,103 @@ def generateLanguageNameTestDataObjects(self, rawtestdata, json_tests, json_veri
...
+            # Ignore blank and # comment lineslines()
+            if len(line) == 0 or line[0] == "#":
+                continue
+            # split at ";" and ignore whitespace
+            tags = list(map(str.strip, line.split(';')))
+            # This should give 3 items
+            if line.find('FAIL') >= 0:
+                # What do do with this?
+                pass
+
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply testgen/testdata_gen.py:212]: Removed this, using "FAIL" as if it's expected data.

File: executors/node/likely_subtags.js (hunk 3 above)
Comment (sven-oly) [reply executors/node/likely_subtags.js:33]: Updated now.

File: testgen/testdata_gen.py (hunk 4 above)
Comment (sven-oly) [reply testgen/testdata_gen.py]: Simplified, creating maximum of 3 cases per source locale.

File: testgen/testdata_gen.py [hunk 6]
```diff
@@ -172,6 +172,103 @@ def generateLanguageNameTestDataObjects(self, rawtestdata, json_tests, json_veri
...
+            # Create minimize tests - default is RemoveFavorScript
+            if tags[2]:
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply testgen/testdata_gen.py]: Done

**Comment by sven-oly** [comment]
Much better now! PTAL

File: executors/node/likely_subtags.js (hunk 3 above)
Comment (sffc) [reply executors/node/likely_subtags.js:33]: I don't see where this is fixed but OK to fix later

**Review by sffc — APPROVED** [review]

**Pull request merged**